dpgen2/exploration/deviation/deviation_std.py

```diff
@@ -0,0 +1,49 @@
+from collections import (
+    defaultdict,
+)
+from typing import (
+    Dict,
+    List,
+    Optional,
+)
+
+import numpy as np
+
+from .deviation_manager import (
+    DeviManager,
+)
+
+
+class DeviManagerStd(DeviManager):
+    r"""The class which is responsible for model deviation management.
+
+    This is the standard implementation of DeviManager. Each deviation
+    (e.g. max_devi_f, max_devi_v in file `model_devi.out`) is stored
+    as a List[Optional[np.ndarray]], where np.array is a one-dimensional
+    array.
+    A List[np.ndarray][ii][jj] is the force model deviation of the jj-th
+    frame of the ii-th trajectory.
+    The model deviation can be List[None], where len(List[None]) is
+    the number of trajectory files.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._data = defaultdict(list)
+
+    def _add(self, name: str, deviation: np.ndarray) -> None:
```

Comment:
Adding frames one by one may be dangerous. The user is allowed to add to different items different numbers of frames, then the frames will never be aligned. 

I would suggest adding all the frames once, i.e. deviation is List[np.ndarray] or a np.ndarray of dim==2

Reply:
@wanghan-iapcm How about adding a `check_data` function?

Reply:
@wanghan-iapcm How about adding a `check_data` function like this:

Reply:
```
@abstractmethod
def check_data(self):
    r""" Check if data is valid
    """
    pass
```
one must call `check_data` function before get data from DeviManager.